ext/AbstractFFTsChainRulesCoreExt.jl

```diff
@@ -37,7 +37,9 @@ function ChainRulesCore.rrule(::typeof(rfft), x::AbstractArray{<:Real}, dims)
 
     project_x = ChainRulesCore.ProjectTo(x)
     function rfft_pullback(ȳ)
-        x̄ = project_x(brfft(ChainRulesCore.unthunk(ȳ) ./ scale, d, dims))
+        ybar = ChainRulesCore.unthunk(ȳ)
+        _scale = convert(typeof(ybar),scale)
```

Comment:
I'm confused by this line.  Isn't `ybar` a vector and `scale` is a scalar?

Reply:
`scale` is an array computed a few lines above:

```julia
    scale = reshape(
        [i == 1 || (i == n && 2 * (i - 1) == d) ? 1 : 2 for i in 1:n],
        ntuple(i -> i == first(dims) ? n : 1, Val(ndims(x))),
    )
```

I didn't care to touch that because it was already there, though if I'm reading it correctly `ntuple(i -> i == first(dims) ? n : 1, Val(ndims(x)))` is `(n, 1, 1, ..., 1)` and so in broadcast the reshape is effectively nothing and it's just a vector of 1's and 2's defined by `[i == 1 || (i == n && 2 * (i - 1) == d) ? 1 : 2 for i in 1:n]` with trailing dimensions 1 (which broadcast would handle...).